src/theory/quantifiers/sygus/sygus_unif_rl.h

```diff
@@ -300,6 +300,63 @@ class SygusUnifRl : public SygusUnif
      private:
       /** reference to parent unif util */
       DecisionTreeInfo* d_dt;
+      /** true and false nodes */
+      Node d_true;
+      Node d_false;
+      /** cache of conditions evaluations on heads */
+      std::map<std::pair<Node, Node>, Node> d_eval_cond_hd;
+
+      /** rebuild decision tree using information gain heuristic
+       *
+       * In a scenario in which the decision tree potentially contains more
+       * conditions than necessary, it is beneficial to rebuild it in a way that
+       * "better" conditions occurr closer to the top.
+       *
+       * The information gain heuristic selects conditions that lead to a
+       * greater reduction of the Shannon entropy in the set of points
+       */
+      void recomputeSolHeuristically(std::map<Node, Node>& hd_mv);
```

Comment:
It seems that PointSeparator is a low level utility, shouldn't this function be at the same level as buildSolution in DecisionTreeInfo?

Reply:
The same goes for buildDtbuildDtInfoGain.